src/Orleans.Serialization/Buffers/Writer.cs

```diff
@@ -349,8 +349,23 @@ private void WriteMultiSegment(scoped ReadOnlySpan<byte> source)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void WriteByte(byte value)
         {
-            EnsureContiguous(1);
-            _currentSpan[_bufferPos++] = value;
+            if ((uint)_bufferPos < (uint)_currentSpan.Length)
+            {
+                Unsafe.AddByteOffset(ref MemoryMarshal.GetReference(_currentSpan), (uint)_bufferPos) = value;
```

Comment:
This is faster than indexing? It seems like something the JIT team should look into, if so